pkg/util/azureclient/azuresdk/armnetwork/interfaces.go

```diff
@@ -0,0 +1,41 @@
+package armnetwork
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+)
+
+// InterfacesClient is a minimal interface for azure InterfacesClient
+type InterfacesClient interface {
+	InterfacesClientAddons
+	Get(ctx context.Context, resourceGroupName string, networkInterfaceName string, options *armnetwork.InterfacesClientGetOptions) (result armnetwork.InterfacesClientGetResponse, err error)
+}
+
+type interfacesClient struct {
+	*armnetwork.InterfacesClient
+}
+
+var _ InterfacesClient = &interfacesClient{}
+
+// NewInterfacesClient creates a new InterfacesClient
+func NewInterfacesClient(environment *azureclient.AROEnvironment, subscriptionID string, credential azcore.TokenCredential) (InterfacesClient, error) {
```

Comment:
We only pass the FPCredential above. Does this expect the FPCredential always?

Reply:
The question is whether `credential` can be different from `fpCredential`?

It is always `fpCredential` for `LoadBalancersClient` and `InterfacesClient` as the previous implementation always uses fpAuthorizer.
This `credential` is a replacement of `authorizer` for the track1.
https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/MIGRATION_GUIDE.md#authentication

When we replace the SDK of `fpPrivateEndpoints`, `rpPrivateLinkServices`, or `dns`, the credential should be different from `fpCredential` because it uses the different authorizer (see `pkg/cluster/cluster.go`).

Reply:
Cool. Then if I understood you correctly we should rename the variable to fpCredential. 

As a separate future PR I think we could extend  azidentity.ClientCertificateCredential to ServicePrincipalCredential and FirstPartyCredential to give us some more assurance and type-safety. What do you think?

Reply:
Mostly I agree with the idea.
My concern is that MIWI feature might use the different credential type.
In that case, we should keep it as it is, but I'm not sure about it though.

Reply:
So my initial answer was incorrect. sorry about that.
> It is always fpCredential for LoadBalancersClient and InterfacesClient as the previous implementation always uses fpAuthorizer.

This is correct just for now.
This can be `WorkloadIdentityCredential`, maybe, in the future.
not sure what the implementation would be like though

https://docs.google.com/document/d/1AIuf6H9y0Jjboywp0NgD8-9lU1jCj5WrE83jh58rEg4/edit
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore@v1.11.1/internal/exported#TokenCredential
https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#WorkloadIdentityCredential